gcc/rust/hir/tree/rust-hir-expr.h

```diff
@@ -2370,6 +2370,14 @@ class FieldAccessExpr : public ExprWithoutBlock
 
   void accept_vis (HIRVisitor &vis) override;
 
+  std::unique_ptr<Expr> &get_receiver_expr ()
```

Comment:
This is a copy paste from the AST::FieldAccessExpr. In each of the passes I am passing around raw pointers but i think it would be better to pass std::unique_ptr<Expr>& so it implies there is no ownership of the pointer being passed around. Though that is something I want to look at fixing as part of 'as' expressions which is the last part of the TyTy module that needs some design.